python/sbp/client/loggers/device_iterator.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (C) 2015 Swift Navigation Inc.
+# Contact: Mark Fine <mark@swiftnav.com>
+#
+# This source is subject to the license found in the file 'LICENSE' which must
+# be be distributed together with this source. All other rights reserved.
+#
+# THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+# EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+
+from base_logger import LogIterator
+from ...client.handler import Handler
+from ...table import dispatch
+import calendar
+import time
+from Queue import Queue, Empty
+
+
+
+class SbpQueue(object):
+  """
+  SbpQueue
+
+  The :class: `SbpQueue` provides a wrapper around Pythons Queue
+
+  Parameters
+  ----------
+  handler : Handler
+    handler object for SBP messages
+  maxsize : int
+    maxsize of the Queue object (0 means infinitie)
+
+  """
+  def __init__(self, handler, maxsize=0 ):
+    self.myqueue = Queue(maxsize)
+
+  def callback(self, msg):
+    """
+    called with each receipt of the message
+    Note: dispatch is necessary but perhaps should be handled in handler
+    """
+    dismsg = dispatch(msg)
+    self.myqueue.put(dismsg, True)
+
+class DeviceIterator(LogIterator):
+  """
+  LogIterator
+
+  The :class: `DeviceIterator` provides an abstract interface for reading
+  from a handle and appending messages to a queue, and reading the queue
+  in an iterable fashion
+
+  Parameters
+  ----------
+  handler : Handler
+    SBP handler
+  timeout : float
+    number of seconds to block when reading queue before
+    raising stop iteration
+
+  """
+  def __init__(self, handler, timeout):
+
+    self.handler = handler
+    self.myqueue = SbpQueue(self.handler)
+    self.handler.add_callback(self.myqueue.callback)
+    self.timeout = timeout
+    self.init_time = time.time()
```

Comment:
Can make this `base_time` and introduce a `delta` method similar to BaseLogger to keep a similar pattern.